components/instructions/programs/associatedTokenAccount.tsx

```diff
@@ -0,0 +1,50 @@
+import { Connection, PublicKey } from '@solana/web3.js'
+import { AccountMetaData } from '@solana/spl-governance'
+import { SPL_TOKENS } from '@utils/splTokens'
+
+const ATA_PROGRAM_ID = new PublicKey(
+  'ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL'
+)
+
+export const ATA_PROGRAM_INSTRUCTIONS = {
+  [ATA_PROGRAM_ID.toBase58()]: {
+    name: 'Associated Token Account Program - Create Associated Token Account',
+    accounts: [
+      'Authority',
+      'Associated Account',
+      'Token Address',
+      'Token Program',
+    ],
+    getDataUI: (
+      _connection: Connection,
+      _data: Uint8Array,
+      accounts: AccountMetaData[]
+    ) => {
+      const ata = accounts[1].pubkey.toString()
+      const tokenMint = accounts[3].pubkey.toString()
+
+      const tokenName = Object.keys(SPL_TOKENS).reduce((tmp, x) => {
```

Comment:
Replace by a `find(...) ?? 'unknown'` ?

Reply:
Object.keys would return the key. So you would have to do something like:

```
const tokenKey = Object.keys(SPL_TOKENS).find(...) 

const tokenName = tokenKey ? SPL_TOKENS[tokenKey].name : "unknown"
```

Reply:
`SPL_TOKENS[Object.keys(SPL_TOKENS).find(...)]?.name ?? 'unknown'` :P

still better than object.keys + reduce imo

Reply:
`Type 'undefined' cannot be used as an index type`

```suggestion
       const tokenName =
        SPL_TOKENS[
          Object.keys(SPL_TOKENS).find(
            (name) => SPL_TOKENS[name].mint?.toString() === tokenMint
          )!
        ].name ?? 'unknown'
```

Reply:
ok nevermind